### PR TITLE
define DLT_PATH_MAX for max path buffer length

### PR DESCRIPTION
**Pull request by votrungchi**
define DLT_PATH_MAX for max path buffer length

DLT limits the path length and does not do anything else to determine
the actual value, because the least that is supported on any system
that DLT runs on is 1024 bytes.

Signed-off-by: Vo Trung Chi <Chi.VoTrung@vn.bosch.com>

### CODE REVIEW

**Comment by sherkenhoff** [comment]
> Since POSIX isn’t clear on path lengths, use PATH_MAX as a base for path + filename.

In POSIX, we can use [pathconf](http://pubs.opengroup.org/onlinepubs/009695399/functions/pathconf.html) with _PC_PATH_MAX - just verified on QNX (returns 1024 bytes) and on Linux (returns 4096 bytes). Is there any reason to not use this interface to determine the _proper_ value?

**Comment by votrungchi** [comment]
Hi @sherkenhoff,

First, we need this value at the compile time, so if we can use the pathconf, it would be use in compile time.
I read about the PATH_MAX and pathconf in this [blog](https://eklitzke.org/path-max-is-tricky). Honestly, I not really sure about the idea of section "Path Metadata With pathconf()", what do you think?

Thanks & Best Regards,

**Comment by sherkenhoff** [comment]
@votrungchi 

This is the same source I also read. Further down it explicitly states that the maximum length of a filename is eventually governed by the filesystem that is handling the node, and that PATH_MAX is only a lower bound, at best. With this in mind, I would not limit the length at compile time to the target's lower bound, but rather determine the actual value at run time at the very little cost of of the pathconf call. 

What would need to be changed in DLT to achieve determining the max path length at run time?

**Comment by votrungchi** [comment]
Hi @sherkenhoff ,

Thank you for your detail explanation.
As my understand, we will use pathconf to determining the max path length at run time with path input is root directory "/". Is this correct?

For your question, to achieve determining the max path length at run time, will have to change all static memory allocation which size is PATH_MAX to dynamic memory allocation.

Best Regards,

**Comment by sherkenhoff** [comment]
>we will use pathconf to determining the max path length at run time with path input is root directory "/"

No. "/" itself might also be mounted and handled by a filesystem that supports much longer names. If that were the case and somebody wanted to write the logs to a very nested location, he couldn't do it because DLT will now set the max length to 1024.

Adding to my question from before: How much load on the system would we introduce by changing to run-time determining the max length? I assume this happens once per process and once in the daemon?

File: include/dlt/dlt_common.h [hunk 1]
```diff
@@ -377,6 +377,11 @@ enum {
  */
 #   define DLT_LINE_LEN 1024
 
+/**
+ * DLT fifo basedir length
+ */
+#define DLT_BASEDIR_MAX PATH_MAX - 16
```
Comment (ssugiura) [inline include/dlt/dlt_common.h]: Where did 16 come from? Either having another macro or give some explanation would be helpful.

File: src/dbus/dlt-dbus-options.c [hunk 2]
```diff
@@ -30,6 +30,7 @@
 
 #include <stdlib.h>
 #include <string.h>
+#include <getopt.h>
```
Comment (thanhbnq) [inline src/dbus/dlt-dbus-options.c]: This header is not really needed here. Please don't add it.

Comment (votrungchi) [reply src/dbus/dlt-dbus-options.c]: Removed. Thanks.

**Pull request merged**